pkg/gofr/datasource/file/ftp/file.go

```diff
@@ -0,0 +1,299 @@
+		f.logger.Errorf("ReadAll Failed. Unable to read json file: %v", err)
+		return nil, err
+	}
+
+	reader := bytes.NewReader(buffer)
+
+	decoder := json.NewDecoder(reader)
+
+	// Peek the first JSON token to determine the type
+	token, err := decoder.Token()
+	if err != nil {
+		return nil, err
+	}
+
+	if d, ok := token.(json.Delim); ok && d == '[' {
+		return &jsonReader{decoder: decoder, token: token}, err
+	}
+
+	// Reading JSON object
+	decoder = json.NewDecoder(reader)
+
+	return &jsonReader{decoder: decoder}, nil
+}
+
+// createTextCSVReader creates a text reader for reading text files.
+func (f *file) createTextCSVReader() (container.RowReader, error) {
+	defer f.postProcess(&FileLog{Operation: "Text/CSV Reader", Location: f.path}, time.Now())
+
+	res, err := f.conn.Retr(f.path)
+	if err != nil {
+		f.logger.Errorf("ReadAll failed. Unable to retrieve text file: %v", err)
+		return nil, err
+	}
+
+	defer res.Close()
+
+	buffer, err := io.ReadAll(res)
+	if err != nil {
+		f.logger.Errorf("ReadAll failed. Unable to read text file: %v", err)
+		return nil, err
+	}
+
+	reader := bytes.NewReader(buffer)
+
+	return &textReader{
+		scanner: bufio.NewScanner(reader),
+		logger:  f.logger,
+	}, err
+}
+
+// Next checks if there is another JSON object available.
+func (j *jsonReader) Next() bool {
+	return j.decoder.More()
+}
+
+// Scan decodes the next JSON object into the provided structure.
+func (j *jsonReader) Scan(i interface{}) error {
+	return j.decoder.Decode(&i)
+}
+
+// Next checks if there is another line available in the text file.
+func (f *textReader) Next() bool {
+	return f.scanner.Scan()
+}
+
+// Scan scans the next line from the text file into the provided pointer to string.
+func (f *textReader) Scan(i interface{}) error {
+	if val, ok := i.(*string); ok {
+		*val = f.scanner.Text()
+		return nil
+	}
+
+	return errNotPointer
+}
+
+// Close closes the FTP file connection.
+func (f *file) Close() error {
+	defer f.postProcess(&FileLog{Operation: "Close", Location: f.path}, time.Now())
+
+	return f.response.Close()
+}
+
+// Name returns the name of the file.
+func (f *file) Name() string {
+	defer f.postProcess(&FileLog{Operation: "Get Name", Location: f.path}, time.Now())
+
+	return f.name
+}
+
+// Read reads data from the FTP file into the provided byte slice and updates the file offset.
+func (f *file) Read(p []byte) (n int, err error) {
+	defer f.postProcess(&FileLog{Operation: "Read", Location: f.path}, time.Now())
+
+	r, err := f.conn.RetrFrom(f.path, uint64(f.offset))
+	if err != nil {
+		f.logger.Errorf("Read failed : Failed to open file with path %q : %v", f.path, err)
+		return 0, err
+	}
+
+	defer r.Close()
+
+	n, err = r.Read(p)
+
+	f.offset += int64(n)
+
+	if err != nil && !errors.Is(err, io.EOF) {
+		f.logger.Errorf("Read failed : Failed to read from %q : %v", f.path, err)
+		return n, err
+	}
+
+	f.logger.Logf("Read successful : Read %v bytes from %q", n, f.path)
+
+	if errors.Is(err, io.EOF) {
+		return n, io.EOF
```

Comment:
Shouldn't it be?

```suggestion
		return n, err
```

Or better 

```suggestion
		return n, nil
```

Or simply remove the if block ?